pkg/lso/localvolumediscovery.go

```diff
@@ -0,0 +1,270 @@
+func NewLocalVolumeDiscoveryBuilder(apiClient *clients.Settings, name, nsname string) *LocalVolumeDiscoveryBuilder {
+	glog.V(100).Infof("Initializing new localVolumeDiscovery structure with the following params: name: "+
+		"%s, namespace: %s", name, nsname)
+
+	builder := LocalVolumeDiscoveryBuilder{
+		apiClient: apiClient,
+		Definition: &lsoV1alpha1.LocalVolumeDiscovery{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: nsname,
+			},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the localVolumeDiscovery is empty")
+
+		builder.errorMsg = "localVolumeDiscovery 'name' cannot be empty"
+	}
+
+	if nsname == "" {
+		glog.V(100).Infof("The nsname of the localVolumeDiscovery is empty")
+
+		builder.errorMsg = "localVolumeDiscovery 'nsname' cannot be empty"
+	}
+
+	return &builder
+}
+
+// PullLocalVolumeDiscovery retrieves an existing localVolumeDiscovery object from the cluster.
+func PullLocalVolumeDiscovery(apiClient *clients.Settings, name, nsname string) (*LocalVolumeDiscoveryBuilder, error) {
+	glog.V(100).Infof(
+		"Pulling localVolumeDiscovery object name:%s in namespace: %s", name, nsname)
+
+	builder := LocalVolumeDiscoveryBuilder{
+		apiClient: apiClient,
+		Definition: &lsoV1alpha1.LocalVolumeDiscovery{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: nsname,
+			},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the localVolumeDiscovery is empty")
+
+		builder.errorMsg = "localVolumeDiscovery 'name' cannot be empty"
+	}
+
+	if nsname == "" {
+		glog.V(100).Infof("The namespace of the localVolumeDiscovery is empty")
+
+		builder.errorMsg = "localVolumeDiscovery 'nsname' cannot be empty"
+	}
+
+	if !builder.Exists() {
+		return nil, fmt.Errorf("localVolumeDiscovery oject %s doesn't exist in namespace %s", name, nsname)
+	}
+
+	builder.Definition = builder.Object
+
+	return &builder, nil
+}
+
+// Get fetches existing localVolumeDiscovery from cluster.
+func (builder *LocalVolumeDiscoveryBuilder) Get() (*LocalVolumeDiscoveryBuilder, error) {
+	if valid, err := builder.validate(); !valid {
+		return nil, err
+	}
+
+	glog.V(100).Infof("Pulling existing localVolumeDiscovery with name %s under namespace %s from cluster",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	lvd := &lsoV1alpha1.LocalVolumeDiscovery{}
+	err := builder.apiClient.Get(context.TODO(), goclient.ObjectKey{
+		Name:      builder.Definition.Name,
+		Namespace: builder.Definition.Namespace,
+	}, lvd)
+
+	if err != nil {
+		return nil, err
+	}
+
+	builder.Object = lvd
+
+	return builder, nil
+}
+
+// Exists checks whether the given localVolumeDiscovery exists.
+func (builder *LocalVolumeDiscoveryBuilder) Exists() bool {
+	if valid, _ := builder.validate(); !valid {
+		return false
+	}
+
+	glog.V(100).Infof("Checking if localVolumeDiscovery %s exists in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	_, err := builder.Get()
```

Comment:
Please replace it by:

builder.Object, err = builder.Get()

Reply:
I'm not interested in the builder.Object here, it will be unused

Reply:
Even if it's going to be unused let's follow standard object structure.